### PR TITLE
[SPARK-10863][SPARKR] Method coltypes() (New version)

### PR DESCRIPTION
**Pull request by olarayej**
This is a follow up on PR #8984, as the corresponding branch for such PR was damaged.

### CODE REVIEW

**Comment by yu-iskw** [comment]
@olarayej could you please check the style warnings with `./dev/lint-r` on your local computer before pushing commits. Thanks.

File: R/pkg/R/schema.R [hunk 1]
```diff
@@ -115,20 +115,7 @@ structField.jobj <- function(x) {
...
-                     "character",
-                     "string",
-                     "binary",
-                     "raw",
-                     "logical",
-                     "boolean",
-                     "timestamp",
-                     "date")
-  if (type %in% primtiveTypes) {
+  if (type %in% names(PRIMITIVE_TYPES)) {
```
Comment (felixcheung) [inline R/pkg/R/schema.R]: To avoid search, `if (!is.null(PRIMITIVE_TYPES[[type]]))`

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -2152,3 +2152,47 @@ setMethod("with",
...
             eval(substitute(expr), envir = newEnv, enclos = newEnv)
           })
+
+#' Returns the column types of a DataFrame.
+#' 
+#' @name coltypes
+#' @title Get column types of a DataFrame
+#' @param x (DataFrame)
+#' @return value (character) A character vector with the column types of the given DataFrame
+#' @rdname coltypes
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: Please add family
```
#' @family dataframe_funcs
```

and example, like
```
#' @examples
#' \dontrun{
#' with(irisDf, nrow(Sepal_Width))
#' }
```

**Comment by felixcheung** [comment]
doc comment, looks good to me otherwise.
@sun-rui @shivaram

**Comment by sun-rui** [comment]
LGTM

**Comment by olarayej** [comment]
I have addressed the docs suggestions. Thank you folks.

**Comment by olarayej** [comment]
@felixcheung @sun-rui All set. Can we please merge? Thanks!

**Comment by shivaram** [comment]
Thanks @olarayej @sun-rui and @felixcheung. LGTM and merging this

**Pull request closed without merging**